extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -233,11 +234,13 @@ void generateDataSourceSupportBean(AgroalRecorder recorder,
             return;
         }
 
-        // make a DataSourceProducer bean
+        // make a DataSources bean
         additionalBeans.produce(AdditionalBeanBuildItem.builder().addBeanClasses(DataSources.class).setUnremovable()
                 .setDefaultScope(DotNames.SINGLETON).build());
         // add the @DataSource class otherwise it won't be registered as a qualifier
         additionalBeans.produce(AdditionalBeanBuildItem.builder().addBeanClass(DataSource.class).build());
+        // make source datasources are initialized at startup
```

Comment:
Too late now, but I am pretty sure you meant to say `make sure`

Reply:
Raaaah. Fixed in https://github.com/quarkusio/quarkus/pull/35273 .